unicodetools/src/main/java/org/unicode/text/UCA/WriteCollationData.java

```diff
@@ -1887,177 +1876,45 @@ public static UCA getCollator(CollatorType type) {
-            CharSequence rep2 = filter(repChar);
-            if (rep2 == null) {
-                rep2 = repChar;
-                fractionalLog.println(
-                        "# Warning - No NFKD primary with:\t"
-                                + Utility.hex(ducetPrimary)
-                                + "\t"
-                                + repChar
-                                + "\t"
-                                + Default.ucd().getCodeAndName(repChar.toString()));
-                // continue;
-            }
-            rep2 = repChar;
-            final int firstChar = Character.codePointAt(rep2, 0);
+            final int firstChar = Character.codePointAt(repChar, 0);
             final int cat = Default.ucd().getCategory(firstChar);
             switch (cat) {
-                case UCD_Types.SPACE_SEPARATOR:
-                case UCD_Types.LINE_SEPARATOR:
-                case UCD_Types.PARAGRAPH_SEPARATOR:
-                case UCD_Types.CONTROL:
-                    spaces.add(ducetPrimary);
-                    break;
                 case UCD_Types.DASH_PUNCTUATION:
                 case UCD_Types.START_PUNCTUATION:
                 case UCD_Types.END_PUNCTUATION:
                 case UCD_Types.CONNECTOR_PUNCTUATION:
                 case UCD_Types.OTHER_PUNCTUATION:
                 case UCD_Types.INITIAL_PUNCTUATION:
                 case UCD_Types.FINAL_PUNCTUATION:
-                    punctuation.add(ducetPrimary);
-                    break;
-                case UCD_Types.DECIMAL_DIGIT_NUMBER:
-                    numbers.add(ducetPrimary);
-                    break;
-                case UCD_Types.LETTER_NUMBER:
-                case UCD_Types.OTHER_NUMBER:
-                    if (ducetPrimary >= firstScriptPrimary) {
-                        break;
-                    }
-                    numbers.add(ducetPrimary);
-                    break;
-                case UCD_Types.CURRENCY_SYMBOL:
-                    currencySymbols.add(ducetPrimary);
-                    break;
-                case UCD_Types.MATH_SYMBOL:
-                case UCD_Types.MODIFIER_SYMBOL:
-                case UCD_Types.OTHER_SYMBOL:
-                    generalSymbols.add(ducetPrimary);
-                    break;
-                case UCD_Types.UNASSIGNED:
-                case UCD_Types.UPPERCASE_LETTER:
-                case UCD_Types.LOWERCASE_LETTER:
-                case UCD_Types.TITLECASE_LETTER:
-                case UCD_Types.MODIFIER_LETTER:
-                case UCD_Types.OTHER_LETTER:
-                case UCD_Types.NON_SPACING_MARK:
-                case UCD_Types.ENCLOSING_MARK:
-                case UCD_Types.COMBINING_SPACING_MARK:
-                case UCD_Types.FORMAT:
-                    if (ducetPrimary >= firstScriptPrimary) {
-                        break;
+                    if (ducetPrimary <= ducetVariableHigh && ducetPrimary > cldrVariableHigh) {
```

Comment:
FYI: We no longer reorder, but we need to find the last punctuation character for CLDR's tailored (lower) variable high primary.